code/obj/effects/foam.dm

```diff
@@ -67,7 +67,9 @@
 	//playsound(src, "sound/effects/bubbles2.ogg", 80, 1, -3)
 
 	update_icon()
-
+	if(metal)
+		if(istype(loc, /turf/space))
```

Comment:
Why not type cast instead of using : if you're already checking the type?

Reply:
copy and pasted from metal foam - could be changed easily enough